modules/gateways/blockonomics/blockonomics.php

```diff
@@ -949,4 +1078,62 @@ public function get_order_checkout_params($params)
 
         return $order_params;
     }
-}
+
+    public function start_polling_job() {
+        $disabled_functions = ini_get('disable_functions');  
+        
+        if (strpos($disabled_functions, 'exec') !== false) {
+            if (function_exists('logModuleCall')) {
+                logModuleCall(
+                    "Blockonomics",
+                    "Validation",
+                    "To check Exec working or not",
+                    "The background job does not work or is disabled. Please check the server configuration"
+                );
+            }
+            return;
+        }
+
+        $path = __DIR__ . '/pollTransactionStatus.php';
+
+        // Command to check if the specific PHP script is running
+        $checkCommand = "pgrep -f 'php $path'";
```

Comment:
I experience issues with this check. $output is set, but the script is not running in the background. I had to use the following in order for the script to start running correctly:
```
        // Command to check if the specific PHP script is running
        $output = [];
        $checkCommand = "pgrep -f 'php $path'";
        exec($checkCommand, $output, $return_var);

        $pid = (int)$output[0];
        $command = "ps -p $pid";
        $output = [];
        exec($command, $output);
        $isRunning = count($output) > 1;

        if (!$isRunning) {
            $startCommand = "php $path > /dev/null &";
            exec($startCommand, $startOutput, $startReturnVar);
        }
```

Please verify the commands that check if background polling is already running are correct and update accordingly.
Another working alternative:
```
        // Command to check if the specific PHP script is running
        $command = "ps aux | grep $path";
        $output = [];
        exec($command, $output);
        $isRunning = count($output) > 2;

        if (!$isRunning) {
            $startCommand = "php $path > /dev/null &";
            exec($startCommand, $startOutput, $startReturnVar);
        }
```